transfer/adapterbase.go

```diff
@@ -124,11 +124,18 @@ func (a *adapterBase) worker(workerNum int, ctx interface{}) {
 		}
 		tracerx.Printf("xfer: adapter %q worker %d processing job for %q", a.Name(), workerNum, t.Object.Oid)
 
+		// tt is the time that we are to compare the transfer's
+		// `expired_at` property against.
+		tt := time.Now().Add(-objectExpirationGracePeriod)
```

Comment:
You've used `-objectExpirationGracePeriod`, but it should be a positive offset as it was before. I think the naming might have misled you; it's not really a 'grace period', it's trying to account for the fact that actually using the link will take a little time and so if it expires in the very near future it's not worth trying to use it.